### PR TITLE
apm-server tests

### LINKED ISSUE #890: Add e2e testsuite for Cloud managed Elastic Agent setup
Once the cloud specific logic is added to the elastic-agent `container` command (https://github.com/elastic/beats/issues/23697) new e2e tests for this "hybrid" docker image are necessary. 

## What to test
* Legacy mode: APM Server is started according to `apm-server.yml` file if present and Elastic Agent is started with Fleet Server
* ~Otherwise: Elastic Agent is started with Fleet Server and APM Server~

The Elastic Agent takes care of the Kibana setup if necessary, the Fleet Server setup for the cloud managed policy and enrolling itself via Fleet Server. 

The setup takes care via ENV variables and config files. Details to follow when the above linked issue is implemented.

### PR DESCRIPTION
**Pull request by jalvz**
## What does this PR do?

Test the apm server setup used on Cloud 

## Why is it important?

Because it's Cloud

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have run the Unit tests for the CLI, and they are passing locally
- [ ] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

Closes #890

### CODE REVIEW

File: e2e/_suites/fleet/fleet.go [hunk 1]
```diff
@@ -1220,3 +1185,45 @@ func deployAgentToFleet(agentInstaller installer.ElasticAgentInstaller, containe
...
+					},
+				},
+				Vars: map[string]kibana.Var{
+					"period": {
+						Value: "1s",
+						Type:  "string",
+					},
+				},
+			}}
+	case "apm":
```
Comment (mdelapenya) [inline e2e/_suites/fleet/fleet.go:1213]: Super super super nit: can we order the items alphabetically in the map? That would help adding/locating integrations

File: e2e/_suites/fleet/stand-alone.go [hunk 2]
```diff
@@ -26,9 +26,10 @@ import (
 
 // StandAloneTestSuite represents the scenarios for Stand-alone-mode
 type StandAloneTestSuite struct {
-	Cleanup  bool
-	Hostname string
-	Image    string
+	Cleanup     bool
```
Comment (mdelapenya) [inline e2e/_suites/fleet/stand-alone.go:29]: Once merged this change, I'm super tempted to create an issue to merge both test suite structures.

**Comment by jalvz** [comment]
@mdelapenya can you help me understand test failures? there are all in a different suite (metricbeat) and look like this: `No such file or directory: <home>/.op/compose/services/dropwizard/docker-compose.yml`

Replace "dropwizard" which every service (etc). I can reproduce the error locally, but I can also reproduce it in master (with `SUITE="metricbeat" TAGS="dropwizard" LOG_LEVEL=TRACE DEVELOPER_MODE=true make -C e2e functional-test` so I don't know how to go from there.)

**Comment by mdelapenya** [comment]
> @mdelapenya can you help me understand test failures? there are all in a different suite (metricbeat) and look like this: `No such file or directory: <home>/.op/compose/services/dropwizard/docker-compose.yml`
> 
> Replace "dropwizard" which every service (redis, etc). I can reproduce the error locally, but I can also reproduce it in master (with `SUITE="metricbeat" TAGS="dropwizard" LOG_LEVEL=TRACE DEVELOPER_MODE=true make -C e2e functional-test` so I don't know how to go from there.)

Sorry for the back-and-forth with this. Because we use same integrations as in Beats repository, we must sync them for first time from that repo. We have `make -C cli sync-integrations` to do so. Once fetched, you should never need to do that, unless the files are not there.

I know this should be more transparent from development stand point, it's something we are thinking about to change, but we did not want to block the user cloning a big repo such as Beats with certain frequency. Of course, this can be changed from improvement.

**Comment by mdelapenya** [comment]
I thought it was documented for the metricbeat test suite, but it's for Fleet, which is incorrect. We will change it in a new issue

**Comment by jalvz** [comment]
Thanks. I think tests should depend on the state on my machine as least as possible, and should be as self contained as possible (`make test` should work or fail on its own, not depending on which other commands did or did not run before it).

I'll open an issue for that.

**Review by adam-stokes — APPROVED** [review]

**Pull request merged**